exporter/datadogexporter/internal/hostmetadata/internal/ec2/ec2.go

```diff
@@ -68,19 +68,37 @@ func GetHostInfo(logger *zap.Logger) (hostInfo *HostInfo) {
 	defer cancel()
 
 	meta := ec2metadata.New(sess, &aws.Config{
-		Retryer: override.IMDSRetryer,
+		Retryer:                   override.IMDSRetryer,
+		EC2MetadataEnableFallback: aws.Bool(false),
+	})
+
+	metaFallbackEnable := ec2metadata.New(sess, &aws.Config{
+		Retryer:                   override.IMDSRetryer,
+		EC2MetadataEnableFallback: aws.Bool(false),
 	})
 
 	if idDoc, err := meta.GetInstanceIdentityDocumentWithContext(ctx); err == nil {
 		hostInfo.InstanceID = idDoc.InstanceID
 	} else {
-		logger.Warn("Failed to get EC2 instance id document", zap.Error(err))
+		childCtxFallbackEnable, cancelRetryEnable := context.WithTimeout(ctx, override.TimePerCall)
+		defer cancelRetryEnable()
+		if idDocInner, errInner := meta.GetInstanceIdentityDocumentWithContext(childCtxFallbackEnable); errInner == nil {
+			hostInfo.InstanceID = idDocInner.InstanceID
+		} else {
+			logger.Warn("Failed to get EC2 instance id document", zap.Error(err))
+		}
 	}
 
 	if ec2Hostname, err := meta.GetMetadataWithContext(ctx, "hostname"); err == nil {
 		hostInfo.EC2Hostname = ec2Hostname
 	} else {
-		logger.Warn("Failed to get EC2 hostname", zap.Error(err))
+		childCtxFallbackEnable, cancelRetryEnable := context.WithTimeout(ctx, override.TimePerCall)
+		defer cancelRetryEnable()
+		if ec2HostnameInner, errInner := metaFallbackEnable.GetMetadataWithContext(childCtxFallbackEnable, "hostname"); errInner == nil {
+			hostInfo.EC2Hostname = ec2HostnameInner
+		} else {
+			logger.Warn("Failed to get EC2 hostname", zap.Error(err))
+		}
```

Comment:
Do we really need to change the datadog exporter?
Do we somehow rely on code from this exporter in ours?

Reply:
No we do not. I just did a search for where imds was called and added the code there. I can remove this if need be, but I think it is fine to keep it.